projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -625,10 +634,49 @@ public static function get_cart_checkout_info() {
 					'[woocommerce_checkout]'
 				),
 			);
-
 			set_transient( $transient_name, $info, DAY_IN_SECONDS );
+			return $info;
 		}
 
+		// We will only reach here if the Cart/Checkout templates are in use.
+
+		$cart_template        = null;
+		$checkout_template    = null;
+		$cart_template_id     = null;
+		$checkout_template_id = null;
+		$block_controller     = Automattic\WooCommerce\Blocks\Package::container()->get( Automattic\WooCommerce\Blocks\BlockTemplatesController::class );
+		$templates            = $block_controller->get_block_templates( array( 'cart', 'checkout' ) );
+
+		foreach ( $templates as $template ) {
+			if ( 'cart' === $template->slug ) {
+				$cart_template_id = ( $template->id );
+				continue;
+			}
+			if ( 'checkout' === $template->slug ) {
+				$checkout_template_id = ( $template->id );
+			}
+		}
+
+		// Get the template and its contents from the IDs we found above.
+		if ( function_exists( 'get_block_template' ) ) {
+			$cart_template     = get_block_template( $cart_template_id );
+			$checkout_template = get_block_template( $checkout_template_id );
+		}
+
+		if ( function_exists( 'gutenberg_get_block_template' ) ) {
+			$cart_template     = get_block_template( $cart_template_id );
+			$checkout_template = get_block_template( $checkout_template_id );
+		}
+
+		// Update the info transient with data we got from the templates, if the site isn't using WC Blocks we
+		// won't be doing this so no concern about overwriting.
+		$info = array(
+			'cart_page_contains_cart_block'             => str_contains( $cart_template->content, '<!-- wp:woocommerce/cart' ),
+			'cart_page_contains_cart_shortcode'         => str_contains( $cart_template->content, '[woocommerce_cart]' ),
+			'checkout_page_contains_checkout_block'     => str_contains( $checkout_template->content, '<!-- wp:woocommerce/checkout' ),
+			'checkout_page_contains_checkout_shortcode' => str_contains( $checkout_template->content, '[woocommerce_checkout]' ),
```

Comment:
This should return a 1/0 instead of true/false, returning a boolean causes a no value for false instead of 0.

```suggestion
			'cart_page_contains_cart_block'             => str_contains( $cart_template->content, '<!-- wp:woocommerce/cart' ) ? 1 : 0,
			'cart_page_contains_cart_shortcode'         => str_contains( $cart_template->content, '[woocommerce_cart]' ) ? 1 : 0,
			'checkout_page_contains_checkout_block'     => str_contains( $checkout_template->content, '<!-- wp:woocommerce/checkout' ) ? 1 : 0,
			'checkout_page_contains_checkout_shortcode' => str_contains( $checkout_template->content, '[woocommerce_checkout]' ) ? 1 : 0,
```

Reply:
This matches the behavior of `post_contains_text`

Reply:
Yes please go ahead!